### PR TITLE
WIP add block and report button, waiting on core

### PR DESCRIPTION
**Pull request by chrisnojima**
This is me adding the block and report buttons. waiting on rpc changes from @mmaxim  so i can plumb it through

### CODE REVIEW

File: shared/chat/conversation/block-conversation-warning/container.js [hunk 1]
```diff
@@ -22,16 +22,20 @@ const mapStateToProps = (state: TypedState, {routeProps}: OwnProps) => {
...
   ...stateProps,
   ...dispatchProps,
   ...ownProps,
+  onBlockAndReport: () => {
+    dispatchProps.onBlock(stateProps.conversationIDKey, true)
+    dispatchProps.onBack()
+  },
   onBlock: () => {
-    dispatchProps.onBlock(stateProps.conversationIDKey)
+    dispatchProps.onBlock(stateProps.conversationIDKey, false)
```
Comment (cjb) [inline shared/chat/conversation/block-conversation-warning/container.js]: :sheep: up to you, but perhaps consider switching from positional to object args now that there's more than one arg

File: shared/chat/conversation/block-conversation-warning/index.desktop.js [hunk 2]
```diff
@@ -5,16 +5,17 @@ import {globalMargins, globalStyles} from '../../../styles'
...
     <Text type='Body' style={{marginTop: globalMargins.large}}>You won't see this conversation anymore.</Text>
     <Text type='Body' style={{marginTop: globalMargins.small}}>To unblock it, run:</Text>
     <Text type='Terminal' style={{...globalStyles.selectable, alignSelf: 'center', marginTop: globalMargins.small}}>keybase chat hide -u {participants}</Text>
     <Text type='Body' style={{marginTop: globalMargins.small}}>in the terminal.</Text>
-    <Box style={{...globalStyles.flexBoxRow, marginTop: globalMargins.xlarge}}>
+    <Box style={{...globalStyles.flexBoxColumn, marginTop: globalMargins.xlarge, alignItems: 'center'}}>
       <Button type='Secondary' onClick={onBack} label="No, don't block them" />
-      <Button type='Danger' style={{marginLeft: globalMargins.tiny}} onClick={onBlock} label='Yes, block them' />
+      <Button type='Danger' style={{marginTop: globalMargins.medium}} onClick={onBlock} label='Yes, block them' />
+      <Button type='Danger' style={{marginTop: globalMargins.small}} onClick={onBlockAndReport} label='Yes, block and report them' />
```
Comment (cjb) [inline shared/chat/conversation/block-conversation-warning/index.desktop.js]: How about "Yes, block them and report abuse", to make it clear what the purpose of reporting them is?

Comment (chrisnojima) [inline shared/chat/conversation/block-conversation-warning/index.desktop.js]: @malgorithms you decide what this text is

**Comment by malgorithms** [comment]
LGTM for this first version

**Comment by mmaxim** [comment]
@chrisnojima CORE side is done, you can hook up to calling `SetConversationStatusLocal` with `reported` as the status.

**Pull request closed without merging**